generators/generator-base.js

```diff
@@ -1199,12 +1199,23 @@ module.exports = class extends PrivateBase {
      * @return {object} the composed generator
      */
     composeExternalModule(npmPackageName, subGen, options) {
-        const generatorName = jhipsterUtils.packageNameToNamespace(npmPackageName);
-        const generatorCallback = `${generatorName}:${subGen}`;
-        if (!this.env.get(generatorCallback)) {
-            throw new Error(`Generator ${generatorCallback} isn't registered.`);
+        try {
+            let generatorTocall = path.join(process.cwd(), 'node_modules', npmPackageName, 'generators', subGen);
+            if (!fs.existsSync(generatorTocall)) {
```

Comment:
This `if` will always happen - as the file path to the generator is missing the '.js' at the end.